pkg/kubernetes/standalone/src/cockpit-kube-auth/main.go

```diff
@@ -45,18 +45,18 @@ func jsonError(err error) ([]byte, error) {
 	return json.Marshal(data)
 }
 
-func readData(file *os.File) ([]byte, error) {
+func readData(fd int) ([]byte, error) {
 	result := make([]byte, MAX_BUFFER)
-	n, err := file.Read(result)
-	if err != nil && err != io.EOF {
+	n, err := syscall.Read(fd, result)
```

Comment:
Because the file read waits for a EOF that never comes. So we allocate a buffer for the max size, do one read call on the fd and then only return what was actually read. The read doesn't need to be 65536 bytes.